my_project/tab_select/app_select.py

```diff
@@ -127,15 +132,13 @@ def submitted_data(
         lines = get_data(url_store)
```

Comment:
@yunzhu1229 I identified another problem, here, you are requesting the data and then you are requesting again in the switch_ip_si function. This will slow down the code significantly.

Reply:
I have modified the function switch_si_ip. If the state is 'ip', it only does the conversion and will not request the data. If it is 'si', it will request the data. In this way, it can also solve the problem of changing data back from 'ip' state to 'si' state.

Reply:
see other comment, about this approach

Reply:
I understand your meanings, and I have tried to fix the problem. However, I found it is necessary to create df in the switch_si_ip function, because it is needed to output the df-store and the df-store can only be output once in the program. To avoid requesting the data twice, I have stored the lines as a store, it can save the running time to some extent.

Reply:
okay great, @yunzhu1229 have you pushed the changes to the brach so @danielh7-cs9 and his team can start work on it?